persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -702,14 +702,22 @@ private Cursor loadEventsCursor(final long measurementIdentifier) {
                 // Pause reached: Move geoLocationCursor to the first location of the next sub-track
                 // We do this to ignore locations between pause and resume event (STAD-140)
                 while (location.getTimestamp() < resumeEventTime) {
-                    geoLocationCursor.moveToNext();
+                    if (!geoLocationCursor.moveToNext()) {
+                        // No more GeoLocations
+                        break;
+                    }
+
                     // Load next location to see if it's the first location of the next sub-track
                     location = loadGeoLocation(geoLocationCursor);
                 }
 
-                // We reached the first location of the next sub-track, thus, move pointer back one step
-                // As the next outer while iteration will move the pointer one step forward
-                geoLocationCursor.moveToPrevious();
+                // Check if there is another sub-track after this sub-tracks resume event
+                if (location.getTimestamp() >= resumeEventTime) {
+                    // We reached the first location of the next sub-track, thus, move pointer back one step
+                    // As the next outer while iteration will move the pointer one step forward
+                    geoLocationCursor.moveToPrevious();
```

Comment:
this is a really ugly control flow. Isn't there some better way of doing this?

Reply:
Here is an alternative implementation but it's even worse to understand I think:
https://github.com/cyface-de/android-backend/compare/fix/4.2.3/STAD-171...tmp/alternateImplementation1?expand=1

Another alternative would be to
- first slice the tracks into sub tracks
- then remove the locations between the pause and resume of the sub tracks

Please provide feedback if this should be implemented before a put more time into this.